drivers/sensor/st/lsm6dsv16x/lsm6dsv16x_trigger.c

```diff
@@ -177,16 +177,13 @@ static void lsm6dsv16x_handle_interrupt(const struct device *dev)
 	int ret;
 
 	while (1) {
-		if (IS_ENABLED(CONFIG_LSM6DSV16X_STREAM)) {
-			break;
-		}
-
```

Comment:
This was working for me, but I am using I3C with the IBI payload which will read out the status register in to the IBI TIR payload (possibly clearing the interrupt event?)

This change will make it perform a redunant read of the `status_reg` if it gets the interrupt through I3C. This probably could use some additional `if` checks to make it work for both

Reply:
> This was working for me, but I am using I3C with the IBI payload which will read out the status register in to the IBI TIR payload (possibly already clearing the interrupt event with that payload?)

Yes, when the status register gets read by any mean, the interrupt event is cleared out. So in I3C seems this is not required. I would try to remove it with some better conditions...